test/unit/proxy_test.rb

```diff
@@ -841,6 +841,21 @@ class ProxyConfigAffectingChangesTest < ActiveSupport::TestCase
         proxy.destroy
       end
     end
+
+    test 'update policies config with audit check' do
+      proxy = FactoryBot.create :proxy
+      long_string = "a" * 62 * 1024
+      # long_string = "foo"
+      # TODO serialization ends up with `ActiveModel::Errors` in there for some reason
+      proxy.update(policies_config: [{name: 'cors2', version: '0.0.2', configuration: {foo: long_string}}])
+
+      Proxy.with_synchronous_auditing do
+        proxy.update(policies_config: [{name: 'cors2', version: '0.0.3', configuration: {foo: long_string}}])
+
+        audit = proxy.audits.last
+        assert_equal 'update', audit.action
```

Comment:
Shouldn't we check the actual audited changes to see the difference? I know we would be testing the library in that case, but isn't this test testing the Audited library already?

On the other hand, if I pause the debugger here and type `audit.audited_changes` this is what I get:

```
=> {"policies_config"=>[#<Proxy::PoliciesConfig:0x0000000011fa1980
@policies_config=[#<Proxy::PolicyConfig:0x0000000011fa17a0 @name="cors2", 
@version="0.0.2", @configuration={"foo"=>"aaaaaaaaaaaaaaaaaaaaaaaaaaaa…
```

Is that correct? shouldn't I see the serialized string here, instead of the ruby object `<Proxy::PoliciesConfig:0x0000000011fa1980`?

Reply:
This is what I would like to but this is how audited works presently. Not changeable without changing audited.

Reply:
This may help going forward https://github.com/collectiveidea/audited/pull/664

Reply:
wrt testing the library, this is more testing that this amount of data will be saved. This would fail before we increased the field type to MEDIUMTEXT in mysql.

Reply:
If the test checks the data size to be saved, then the test name should reflect that IMO.
Something like: `update big policy chain with audit check`

Reply:
You are right. btw we rather intrusively monkey patch the auditing process, that's why testing the basic functionality also makes sense. But name will be better to reflect intention here.